### PR TITLE
🔒 : vault integration

### LINKED ISSUE #236: ✨ : Integration with Hashicorp Vault for secrets and securely storing state files
**Is your feature request related to a problem? Please describe.**
Thanks for the nice work!
It would be nice to add integration with `Hashicorp Vault` for storing and obtaining secrets as an `environment` or `values`. And also would be great to use Vault Transit `encryption/decryption` for securely storing state files (possibly some leak of sensitive information via the state file). Can you investigate this?

**Describe the solution you'd like**
`Hashicorp Vault` Secret backend and Transit backend

**Describe alternatives you've considered**
It can by other solution, but `Hashicorp Vault` has native integration with Terraform

**Additional context**
- https://learn.hashicorp.com/vault/getting-started/secrets-engines
- https://learn.hashicorp.com/vault/encryption-as-a-service/eaas-transit

### LINKED ISSUE #286: ✨ : Import Service Account JSON variable
Is your feature request related to a problem? Please describe.
How to import Service Account JSON GCP file to runner ? 
`provider "google" {
  credentials = file location
  project     = "tesringt"
  region      = "australia-southeast1"
}`
GAIA have provide enviroment variabel to expoort CREDENTIAL, but location file json from local to inside container terraform, not found.

How to bind service account json to container terraform via /var/run/docker.sock?

Describe the solution you'd like
Provide variable to attach service account to access resource to cloud.

### PR DESCRIPTION
**Pull request by juwit**
Enables vault integration for :
* encryption of credentials content
* getting aws credentials using the vault aws secret engine

Adds VaultAWSCredentials credentials type.

Also adds a docker-compose example.

resolves #286 
resolves #236

### CODE REVIEW

File: src/main/client/app/pages/credentials/providers/credentials-vault-aws.vue [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    methods: {
+      notEmpty(field) {
+        return typeof field !== 'undefined' && field !== null && field.trim() !== '';
+      },
+    },
+  };
+</script>
+
+<style scoped>
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/providers/credentials-vault-aws.vue:47]: we can remove this empty tag

File: src/test/java/io/gaia_app/credentials/CredentialsServiceIT.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    @MockBean
+    lateinit var encryptionService: EncryptionService
+
+    @Autowired
+    lateinit var vaultTemplate: VaultTemplate
+
+    @Test
+    fun `loadAWSCredentialsFromVault should call vault aws secret engine`(){
+        // given
+        val jsonResponse = """{
```
Comment (cdubuisson) [inline src/test/java/io/gaia_app/credentials/CredentialsServiceIT.kt:43]: we can clean the format here, no ?

File: src/test/java/io/gaia_app/encryption/NoOpEncryptionServiceIT.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package io.gaia_app.encryption
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+//
+//@SpringBootTest(classes = [NoOpEncryptionService::class])
```
Comment (cdubuisson) [inline src/test/java/io/gaia_app/encryption/NoOpEncryptionServiceIT.kt:8]: Is the test class still useful ?

File: src/test/java/io/gaia_app/credentials/CredentialsServiceIT.kt (hunk 2 above)
Comment (juwit) [reply src/test/java/io/gaia_app/credentials/CredentialsServiceIT.kt:43]: do you mean having a better indentation ? yes

File: src/test/java/io/gaia_app/encryption/NoOpEncryptionServiceIT.kt (hunk 3 above)
Comment (juwit) [reply src/test/java/io/gaia_app/encryption/NoOpEncryptionServiceIT.kt]: No it is not.
And another PR will remove the NoOpEncryptionService, to better use inheritance.

**Comment by binlab** [comment]
Thank you for this powerful security improvements!

**Pull request merged**